chore: Normalize transaction controller patch

Gudahtt opened the pull request:
<!--
Thanks for your contribution!

Please ensure that any applicable requirements below are satisfied before submitting this pull request. This will help ensure a quick and efficient review cycle.
-->

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

The transaction controller patch has been re-generated following a set of steps that has been documented in the patch itself. This resolves inconsistencies present in the patch resulting from different workflows being used to update it. This includes no functional changes.

**Issue**

This relates to https://github.com/MetaMask/mobile-planning/issues/1226 (this makes it possible to minimize the diff for the patch changes needed as part of the next controller upgrades)

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

Gudahtt committed:
chore: Normalize transaction controller patch

The transaction controller patch has been re-generated following a set
of steps that has been documented in the patch itself. This resolves
inconsistencies present in the patch resulting from different workflows
being used to update it. This includes no functional changes.

Gudahtt commented on patches/@metamask+transaction-controller+4.0.1.patch:
```diff
@@ -3,10 +3,20 @@ new file mode 100644
 +++ b/node_modules/@metamask/transaction-controller/dist/EtherscanRemoteTransactionSource.d.ts
-@@ -0,0 +1,13 @@
+@@ -0,0 +1,23 @@
++// PATCH GENERATED FROM MetaMask/core branch refactor/transaction-controller-patch-mobile
```
The patch comment has been restored, and more detailed instructions added on how to make further updates

Gudahtt commented on patches/@metamask+transaction-controller+4.0.1.patch:
```diff
@@ -127,14 +137,6 @@ index 0000000..ef97d13
-new file mode 100644
-index 0000000..78f6c0d
---- /dev/null
-+++ b/node_modules/@metamask/transaction-controller/dist/EtherscanRemoteTransactionSource.js.map
```
Some of the sourcemap files had already been removed in the patch, but the rest have now been removed as well for consistency

Gudahtt commented on patches/@metamask+transaction-controller+4.0.1.patch:
```diff
@@ -636,21 +630,22 @@ index 56005aa..e7f8bf1 100644
 +        deviceConfirmedOn?: WalletDevice;
 +        origin?: string;
-+        securityAlertResponse?: Recors<string, unknown>;
++        securityAlertResponse?: Record<string, unknown>;
```
This typo was introduced in a recent manual change ot the patch, along with the change to the function signature and JSDoc. Type-only change since this is in a `.d.ts` file

Gudahtt commented:
Moving into draft because it conflicts with #7209 which is more urgent. After that is merged, I can update the patch and mark this as ready for review

matthewwalsh0 commented on patches/@metamask+transaction-controller+4.0.1.patch:
```diff
@@ -1,12 +1,19 @@
 --- /dev/null
 +++ b/node_modules/@metamask/transaction-controller/dist/EtherscanRemoteTransactionSource.d.ts
-@@ -0,0 +1,13 @@
+@@ -0,0 +1,23 @@
```
Would it be better to put the comment above this line otherwise are we not including this message in the first file that happens to be in the diff, so `EtherscanRemoteTransactionSource.d.ts` currently?

Gudahtt replied:
I would but then it wouldn't be a valid patch file. I tried just to make sure, and sure enough, patch-package fails saying that it can't parse the file. The comment needs to be in the file being patched unfortunately.

Gudahtt committed:
Update patch with incoming transaction changes

Gudahtt committed:
Remove mocks and sourcemaps

Gudahtt commented on patches/@metamask+transaction-controller+4.0.1.patch:
```diff
@@ -1966,7 +1950,7 @@ index 0000000..0155bc1
 +            address,
 +            // Not previously used in mobile due to bug, temporarily disabled to match existing behaviour.
-+            startBlock: null,
++            startBlock: undefined,
```
cc @matthewwalsh0 , does this look OK? This must be a difference between the patch and the changes applied on the core branch. Presumably null and undefined are treated equivalently here though

Gudahtt commented:
Passing E2E smoke tests on iOS: https://app.bitrise.io/build/b5bf427a-a45c-4f21-85bc-c03659cd4cc7 (android is currently not passing on `main` for most builds due to a segfault)

Gudahtt committed:
Update the patch using the new `patch:tx` script

The process for updating the transaction controller patch has been
automated with the `patch:tx` script. This script follows the same
steps I had used, except that the patch comment is now in a separate
file and the instructions have been improved.

Gudahtt commented:
This has been rebased, and updated with the changes from #7226 (which added a script to automate transaction controller patch updates, and improved the patch instructions)

Gudahtt commented:
Passing E2E test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/efba570c-d0f1-477f-a1df-671528a96335

Cal-L reviewed (approved):
LGTM

Gudahtt commented:
Updated E2E test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/3eb9602f-a475-4823-9f3b-6e320935423d
